x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/normalization/rule_actions.ts

```diff
@@ -70,10 +70,12 @@ export const transformFromAlertThrottle = (
   if (legacyRuleActions == null || (rule.actions != null && rule.actions.length > 0)) {
     if (rule.muteAll || rule.actions.length === 0) {
       return NOTIFICATION_THROTTLE_NO_ACTIONS;
-    } else if (
-      rule.notifyWhen === 'onActiveAlert' ||
-      (rule.throttle == null && rule.notifyWhen == null)
-    ) {
+    } else if (rule.notifyWhen == null) {
+      const frequency = rule.actions[0].frequency ?? null;
+      if (!frequency || frequency.notifyWhen !== 'onThrottleInterval' || frequency.throttle == null)
```

Comment:
nit: this functionality could be in an internal separate function to highlight its purpose.